command-utility-tool/src/main/java/com/utility/InputHandler.java

```diff
@@ -0,0 +1,108 @@
+package com.utility;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+/**
+ * This class handles user input and retry 3 times if input is invalid.
+ * If input is valid, then check if the input path and file name is valid
+ * If input bye, then exit the process
+ */
+public class InputHandler {
+
+    /**
+     * This function handles user input and retry 3 times if input is invalid.
+     * @return valid path
+     */
+    public String inputFile(Scanner input) throws IOException {
+        System.out.print("Enter Directory name: ");
+        String dirName = input.nextLine();
+        validateDirPath(dirName);
+
+        System.out.print("Enter file name: ");
+        String fileName = input.nextLine();
+        validateFilePath(dirName, fileName);
+
+        String filePath = filePath(dirName, fileName);
+        System.out.println(filePath);
+        return filePath;
+    }
+
+    /**
+     * Input and validate word
+     * @param input word
+     * @return word
+     */
+    public String inputWord(Scanner input) {
+        System.out.print("Enter word: ");
+        String word = input.nextLine().toLowerCase();
+        word = word.replaceAll("[^a-z]$", "");
+        boolean wordValidated = validateWord(word);
+        while (!wordValidated) {
+            System.out.print("Invalid word, please input again: ");
+            System.out.print("Enter word: ");
+            String newWord = input.nextLine();
+            boolean newValidated = validateWord(newWord);
+            if (newValidated == true) {
+                word = newWord;
+                break;
+            }
```

Comment:
This while loop is only used once when the first word is whether valid or not from line 44 to 45 and there is a new check by if condition from line 50. 

Can it be refactored similar to this?

`  # new method
String wordProcessed(String word) {
   word = input.nextLine().toLowerCase();
   return word.replaceAll("[^a-z]$", "");
}
`

Then use the method for while condition and refactor line 48 to 53?
`while(validateWord(wordProcessed(word)))`

Reply:
good catch